library/ZendService/Twitter/Media.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/ZendService_Twitter for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/ZendService_Twitter/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendService\Twitter;
+
+use Zend\Http\Client as Client;
+use ZendService\Twitter\Response as Response;
+
+/**
+ * Twitter Media Uploader base class
+ *
+ * @author Cal Evans <cal@calevans.com>
+ */
+class Media
+{
+    const UPLOAD_BASE = 'https://upload.twitter.com';
+
+    const UPLOAD_PATH = '/1.1/media/upload.json';
+
+    /**
+     * Array of basic data to be stored by this class.
+     *
+     * @var Array
+     */
+    protected $data = [];
+
+    /**
+     * The number of bytes to send to Twitter at one time.
+     * @var Integer
+     */
+    protected $chunkSize = (1024 * 1024) * 4;
+
+    /**
+     * Constructor
+     *
+     * @param  null|string $imageFilename
+     * @param  string $consumer
+     */
+    public function __construct($imageFilename = null, $media_type = '')
+    {
+        $this->data['image_filename'] = '';
+        $this->data['media_id']       = 0;
+        $this->data['segment_index']  = 0;
+        $this->data['media_type']     = $media_type;
+
+        if (null !== $imageFilename) {
+            $this->data['image_filename'] = $imageFilename;
+        }
+    }
+
+    /**
+     * Main call into the upload workflow
+     *
+     * @param  Client $httpClient
+     * @return Twitter\Response
+     * @throws Exception If the file can't be opened.
+     */
+    public function upload(Client $httpClient)
+    {
+        if (! $this->validateFile($this->data['image_filename'])) {
+            throw new Exception\InvalidMediaException('Failed to open ' . $this->data['image_filename']);
+        }
+
+        if (empty($this->data['media_type'])) {
+            throw new Exception\InvalidMediaException('No Media Type given.');
+        }
+
+        $params = [];
+        $httpClient->setUri(self::UPLOAD_BASE . self::UPLOAD_PATH);
+
+        $params['total_bytes'] = filesize($this->data['image_filename']);
+        $params['media_type']  = $this->data['media_type'];
+
+        $response = $this->initUpload($httpClient, $params);
+        $initResponse = $response->toValue();
```

Comment:
What's the question? This is the response to an `INIT` operation.

The refactor I'm preparing to push renames the variable to simply `$response`, as other responses are not captured, however.